File: android/src/expo/com/smileidentity/expo/views/SmileIDExpoSmartSelfieEnrollmentEnhancedView.kt
package com.smileidentity.expo.views

import android.content.Context
import androidx.compose.runtime.Composable
import com.smileidentity.SmileID
import com.smileidentity.shared.RNSmartSelfieEnrollmentEnhanced
import com.smileidentity.shared.SmileIDViewConfig
import expo.modules.kotlin.AppContext

/**
 * Expo implementation of SmartSelfieEnrollmentEnhanced using the new architecture
 */
class SmileIDExpoSmartSelfieEnrollmentEnhancedView(
    context: Context,
    appContext: AppContext
) : SmileIDExpoView(context, appContext) {

    var allowNewEnroll: Boolean = false

    override fun renderContent() {
        setContentWithTheme {
            RenderSmartSelfieEnrollmentEnhancedContent()
        }
    }

    @Composable
    private fun RenderSmartSelfieEnrollmentEnhancedContent() {
        val config = SmileIDViewConfig(
            userId = userId,
            jobId = jobId,
            allowAgentMode = allowAgentMode ?: false,
            showInstructions = showInstructions,
            skipApiSubmission = skipApiSubmission,
            showAttribution = showAttribution,
            extraPartnerParams = extraPartnerParams,
            allowNewEnroll = allowNewEnroll
        )

        SmileID.RNSmartSelfieEnrollmentEnhanced(
            config = config,
            onResult = { result ->
                handleResultCallback(result)
            }
        )
    }
}

File: package/ios/expo/views/SmileIDExpoSmartSelfieEnrollmentEnhancedView.swift
import ExpoModulesCore
import UIKit
import SwiftUI
import SmileID

/**
 * Expo implementation of SmartSelfieEnrollmentEnhanced using the new architecture
 */
class SmileIDExpoSmartSelfieEnrollmentEnhancedView: SmileIDExpoView {
    
    var allowNewEnroll: Bool = true
    
    override func renderContent() {
        let config = createConfig()
        
        // Use SmileIDExtReactNative to create the view
        let uiView = SmileID.RNSmartSelfieEnrollmentEnhanced(
            config: config,
            onResult: { [weak self] result in
                self?.handleResultCallback(result)
            }
        )
        
        // Use the base class helper to set the UIView
        setUIView(uiView)
    }
    
    private func createConfig() -> SmileIDViewConfig {
        var config = SmileIDViewConfig()
        config.userId = userId
        config.jobId = jobId
        config.allowAgentMode = allowAgentMode
        config.allowNewEnroll = allowNewEnroll
        config.showInstructions = showInstructions
        config.skipApiSubmission = skipApiSubmission
        config.showAttribution = showAttribution
        config.extraPartnerParams = extraPartnerParams
        return config
    }
}